node/src/accountant/payment_adjuster/mod.rs

```diff
@@ -0,0 +1,2340 @@
+        let cw_transaction_fee_minor = gwei_to_wei(cw_balance_transaction_fee_major);
+        let estimated_transaction_fee_per_transaction_minor = gwei_to_wei(
+            estimated_transaction_fee_units_per_transaction * agreed_transaction_fee_price,
+        );
+
+        let blockchain_agent = BlockchainAgentMock::default()
+            .agreed_transaction_fee_margin_result(*TRANSACTION_FEE_MARGIN)
+            .transaction_fee_balance_minor_result(cw_transaction_fee_minor)
+            .service_fee_balance_minor_result(cw_service_fee_balance_minor)
+            .estimated_transaction_fee_per_transaction_minor_result(
+                estimated_transaction_fee_per_transaction_minor,
+            );
+
+        Box::new(blockchain_agent)
+    }
+
+    fn test_inner_was_reset_to_null(subject: PaymentAdjusterReal) {
+        let err = catch_unwind(AssertUnwindSafe(|| {
+            subject.inner.original_cw_service_fee_balance_minor()
+        }))
+        .unwrap_err();
+        let panic_msg = err.downcast_ref::<String>().unwrap();
+        assert_eq!(
+            panic_msg,
+            "Broken code: Broken code: Called the null implementation of \
+        the original_cw_service_fee_balance_minor() method in PaymentAdjusterInner"
+        )
+    }
+
+    // The following tests together prove the use of correct calculators in the production code
+
+    #[test]
+    fn each_of_defaulted_calculators_returns_different_value() {
+        let now = SystemTime::now();
+        let payment_adjuster = PaymentAdjusterReal::default();
+        let qualified_payable = QualifiedPayableAccount {
+            bare_account: PayableAccount {
+                wallet: make_wallet("abc"),
+                balance_wei: gwei_to_wei::<u128, u64>(444_666_888),
+                last_paid_timestamp: now.checked_sub(Duration::from_secs(123_000)).unwrap(),
+                pending_payable_opt: None,
+            },
+            payment_threshold_intercept_minor: gwei_to_wei::<u128, u64>(20_000),
+            creditor_thresholds: CreditorThresholds::new(gwei_to_wei::<u128, u64>(10_000)),
+        };
+        let cw_service_fee_balance_minor = gwei_to_wei::<u128, u64>(3_000);
+        let exceeding_balance = qualified_payable.bare_account.balance_wei
+            - qualified_payable.payment_threshold_intercept_minor;
+        let context = PaymentAdjusterInnerReal::new(
+            now,
+            None,
+            cw_service_fee_balance_minor,
+            exceeding_balance,
+        );
+        let _ = payment_adjuster
+            .calculators
+            .into_iter()
+            .map(|calculator| calculator.calculate(&qualified_payable, &context))
+            .fold(0, |previous_result, current_result| {
+                let min = (current_result * 97) / 100;
+                let max = (current_result * 97) / 100;
+                assert_ne!(current_result, 0);
+                assert!(min <= previous_result || previous_result <= max);
```

Comment:
The two changes that we need here:
- The `min` and `max` should be different (97 and 103)
- We can assert for not in the range by using `assert_ne!(min <= previous_result && previous_result <= max);`